src/main/drivers/dshot.c

```diff
@@ -47,7 +48,6 @@
 
 #include "rx/rx.h"
 
```

Comment:
```suggestion

#include "dshot.h"
```

Reply:
I will fix it in the next push

Reply:
Moving shot header causes automatic unit testing to break

Reply:
Could not fix

Reply:
Then something might be wrong (maybe you need to include dshot.h somewhere else?). What does the failing unit test error say?

Reply:
motor_output_unittest automated unit testing breaks with the error below this explanation.
This is because the place dshot.h header is moved is in a #ifdef USE_DSHOT region, so when compiling dshot.c in motor_output_unittest automated unit testing it breaks, because that automated unit test doesn't have USE_DSHOT defined in src/test/Makefile.
As far as I know there are two possible solutions.
1. Close USE_DSHOT region before including dshot.h and opening it again after
2. Add USE_DSHOT to motor_output_unittest

Please let me know what solution do you prefer

```
compiling ../main/drivers/dshot.c 
../main/drivers/dshot.c:285:16: error: use of undeclared identifier 'DSHOT_TELEMETRY_INVALID'
        return DSHOT_TELEMETRY_INVALID;
               ^
../main/drivers/dshot.c:292:55: error: unknown type name 'dshotTelemetryType_t'                                                      
uint32_t dshot_decode_telemetry_value(uint32_t value, dshotTelemetryType_t *type)
                                                      ^
../main/drivers/dshot.c:310:21: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_STATE_EVENTS'                              
            *type = DSHOT_TELEMETRY_TYPE_STATE_EVENTS;
                    ^
../main/drivers/dshot.c:313:23: error: use of undeclared identifier 'DSHOT_TELEMETRY_INVALID'                                        
            decoded = DSHOT_TELEMETRY_INVALID;
                      ^
../main/drivers/dshot.c:316:21: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_eRPM'                                      
            *type = DSHOT_TELEMETRY_TYPE_eRPM;
                    ^
../main/drivers/dshot.c:329:21: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_TEMPERATURE'                               
            *type = DSHOT_TELEMETRY_TYPE_TEMPERATURE;
                    ^
../main/drivers/dshot.c:337:21: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_VOLTAGE'                                   
            *type = DSHOT_TELEMETRY_TYPE_VOLTAGE;
                    ^
../main/drivers/dshot.c:345:21: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_CURRENT'                                   
            *type = DSHOT_TELEMETRY_TYPE_CURRENT;
                    ^
../main/drivers/dshot.c:353:21: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_DEBUG1'                                    
            *type = DSHOT_TELEMETRY_TYPE_DEBUG1;
                    ^
../main/drivers/dshot.c:361:21: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_DEBUG2'                                    
            *type = DSHOT_TELEMETRY_TYPE_DEBUG2;
                    ^
../main/drivers/dshot.c:369:21: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_DEBUG3'                                    
            *type = DSHOT_TELEMETRY_TYPE_DEBUG3;
                    ^
../main/drivers/dshot.c:377:21: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_STATE_EVENTS'                              
            *type = DSHOT_TELEMETRY_TYPE_STATE_EVENTS;
                    ^
../main/drivers/dshot.c:385:21: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_eRPM'                                      
            *type = DSHOT_TELEMETRY_TYPE_eRPM;
                    ^
../main/drivers/dshot.c:298:10: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_eRPM'                                      
    case DSHOT_TELEMETRY_TYPE_eRPM:
         ^
../main/drivers/dshot.c:303:10: error: use of undeclared identifier 'DSHOT_TELEMETRY_TYPE_STATE_EVENTS'                              
    case DSHOT_TELEMETRY_TYPE_STATE_EVENTS:
         ^
15 errors generated.                                                                                                                 
make: *** [Makefile:775: ../../obj/test/motor_output_unittest/drivers/dshot.c.o] Error 1
```

Reply:
Definitely the second one. There's nothing wrong with adjusting the unittest files.

Reply:
Tryying the second solution adding USE_DSHOT in the unit testing it throws the error below. Function dshotInitEndpoints was in USE_DSHOT region before I made changes. I think it was to avoid dependencies with other modules, not related to motor_output_unittest automated unit test.

Please, let me know if you would accept the proposed solution 1 or please let me know a solution to avoid dependencies.

```
compiling ../main/drivers/dshot.c 
compiling unit/motor_output_unittest.cc 
linking ../../obj/test/motor_output_unittest/motor_output_unittest 
/usr/bin/ld: ../../obj/test/motor_output_unittest/drivers/dshot.c.o: en la función `dshotInitEndpoints':
/home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:54: referencia a `featureIsEnabled' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:55: referencia a `getDigitalIdleOffset' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:57: referencia a `getDigitalIdleOffset' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:60: referencia a `getDigitalIdleOffset' sin definir
/usr/bin/ld: ../../obj/test/motor_output_unittest/drivers/dshot.c.o: en la función `dshotConvertFromExternal':
/home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:71: referencia a `featureIsEnabled' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:75: referencia a `scaleRangef' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:77: referencia a `scaleRangef' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:80: referencia a `scaleRangef' sin definir
/usr/bin/ld: ../../obj/test/motor_output_unittest/drivers/dshot.c.o: en la función `dshotConvertToExternal':
/home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:90: referencia a `featureIsEnabled' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:94: referencia a `scaleRangef' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:96: referencia a `scaleRangef' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/drivers/dshot.c:99: referencia a `scaleRangef' sin definir
/usr/bin/ld: ../../obj/test/motor_output_unittest/drivers/dshot.c.o: en la función `__get_BASEPRI':
/home/ctag/proyectos/betaflight/src/test/../main/build/atomic.h:51: referencia a `atomic_BASEPRI' sin definir
/usr/bin/ld: ../../obj/test/motor_output_unittest/drivers/dshot.c.o: en la función `__basepriRestoreMem':
/home/ctag/proyectos/betaflight/src/test/../main/build/atomic.h:57: referencia a `atomic_BASEPRI' sin definir
/usr/bin/ld: ../../obj/test/motor_output_unittest/drivers/dshot.c.o: en la función `__basepriSetMemRetVal':
/home/ctag/proyectos/betaflight/src/test/../main/build/atomic.h:64: referencia a `atomic_BASEPRI' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/build/atomic.h:64: referencia a `atomic_BASEPRI' sin definir
/usr/bin/ld: /home/ctag/proyectos/betaflight/src/test/../main/build/atomic.h:65: referencia a `atomic_BASEPRI' sin definir
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:777: ../../obj/test/motor_output_unittest/motor_output_unittest] Error 1
```

Reply:
fixed